```
nm -P .libs/compatibility.o .libs/compatibility-debug_list.o .libs/compatibility-debug_list-2.o .libs/compatibility-c++0x.o .libs/compatibility-atomic-c++0x.o .libs/compatibility-thread-c++0x.o .libs/compatibility-chrono.o ../libsupc++/.libs/libsupc++convenience.a ../src/c++98/.libs/libc++98convenience.a ../src/c++11/.libs/libc++11convenience.a|Can't exec "c++filt": No such file or directory at /nix/store/xbbhh106s22rwrchpshkki9j9ah7qbw9-perl-5.20.2/lib/perl5/5.20.2/IPC/Open3.pm line 280.
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
open2: exec of c++filt -_ failed at /private/var/folders/pb/tjgx9w511hjdbh9pld3pf1hr0000gn/T/nix-build-gfortran-4.8.4.drv-0/gcc-4.8.4/libstdc++-v3/scripts/make_exports.pl line 95.
Makefile:889: recipe for target 'libstdc++-symbols.explist' failed
make[6]: *** [libstdc++-symbols.explist] Error 1
make[6]: Leaving directory '/private/var/folders/pb/tjgx9w511hjdbh9pld3pf1hr0000gn/T/nix-build-gfortran-4.8.4.drv-0/build/x86_64-apple-darwin13.4.0/libstdc++-v3/src'
Makefile:575: recipe for target 'all-recursive' failed
make[5]: *** [all-recursive] Error 1
make[5]: Leaving directory '/private/var/folders/pb/tjgx9w511hjdbh9pld3pf1hr0000gn/T/nix-build-gfortran-4.8.4.drv-0/build/x86_64-apple-darwin13.4.0/libstdc++-v3/src'
Makefile:480: recipe for target 'all-recursive' failed
make[4]: *** [all-recursive] Error 1
make[4]: Leaving directory '/private/var/folders/pb/tjgx9w511hjdbh9pld3pf1hr0000gn/T/nix-build-gfortran-4.8.4.drv-0/build/x86_64-apple-darwin13.4.0/libstdc++-v3'
Makefile:386: recipe for target 'all' failed
make[3]: *** [all] Error 2
make[3]: Leaving directory '/private/var/folders/pb/tjgx9w511hjdbh9pld3pf1hr0000gn/T/nix-build-gfortran-4.8.4.drv-0/build/x86_64-apple-darwin13.4.0/libstdc++-v3'
Makefile:11495: recipe for target 'all-stage1-target-libstdc++-v3' failed
make[2]: *** [all-stage1-target-libstdc++-v3] Error 2
make[2]: Leaving directory '/private/var/folders/pb/tjgx9w511hjdbh9pld3pf1hr0000gn/T/nix-build-gfortran-4.8.4.drv-0/build'
Makefile:18418: recipe for target 'stage1-bubble' failed
make[1]: *** [stage1-bubble] Error 2
make[1]: Leaving directory '/private/var/folders/pb/tjgx9w511hjdbh9pld3pf1hr0000gn/T/nix-build-gfortran-4.8.4.drv-0/build'
Makefile:18706: recipe for target 'bootstrap' failed
make: *** [bootstrap] Error 2
builder for ‘/nix/store/5qiysl11y0hl38g7i0p9k56gkyddqcd5-gfortran-4.8.4.drv’ failed with exit code 2
```

This commit does work for me on pure-darwin, just not here.

--- a/pkgs/development/compilers/gcc/gfortran-darwin.nix
+++ b/pkgs/development/compilers/gcc/gfortran-darwin.nix
@@ -1,0 +1,22 @@
+# This is a derivation customized to work on OS X (Darwin).
+{gmp, mpfr, libmpc, fetchurl, stdenv}:
+
+# This package is only intended for OSX.
+assert stdenv.isDarwin;
+
+stdenv.mkDerivation rec {
+  name = "gfortran";
+  version = "5.1.0";
+  buildInputs = [gmp mpfr libmpc];
+  src = fetchurl {
+    url = "https://ftp.gnu.org/gnu/gcc/gcc-${version}/gcc-${version}.tar.bz2";
+    sha256 = "1bd5vj4px3s8nlakbgrh38ynxq4s654m6nxz7lrj03mvkkwgvnmp";
+  };
+  configureFlags = ''
+    --enable-languages=fortran --enable-checking=release --disable-bootstrap
+    --with-gmp=${gmp}
+    --with-mpfr=${mpfr}
+    --with-mpc=${libmpc}
+  '';
+  makeFlags = ["CC=/usr/bin/gcc"];
+}

--- a/pkgs/development/compilers/gcc/gfortran-darwin.nix
+++ b/pkgs/development/compilers/gcc/gfortran-darwin.nix
@@ -18,5 +18,5 @@ stdenv.mkDerivation rec {
     --with-mpfr=${mpfr}
     --with-mpc=${libmpc}
   '';
-  makeFlags = ["CC=/usr/bin/gcc"];
+  makeFlags = ["CC=clang"];
 }

--- a/pkgs/development/compilers/gcc/gfortran-darwin.nix
+++ b/pkgs/development/compilers/gcc/gfortran-darwin.nix
@@ -1,15 +1,13 @@
-# This is a derivation customized to work on OS X (Darwin).
+# This is a derivation specific to OS X (Darwin). It may work on other
+# systems as well but has not been tested.
 {gmp, mpfr, libmpc, fetchurl, stdenv}:
 
-# This package is only intended for OSX.
-assert stdenv.isDarwin;
-
 stdenv.mkDerivation rec {
-  name = "gfortran";
+  name = "gfortran-${version}";
   version = "5.1.0";
   buildInputs = [gmp mpfr libmpc];
   src = fetchurl {
-    url = "https://ftp.gnu.org/gnu/gcc/gcc-${version}/gcc-${version}.tar.bz2";
+    url = "mirror://gnu/gcc/gcc-${version}/gcc-${version}.tar.bz2";
     sha256 = "1bd5vj4px3s8nlakbgrh38ynxq4s654m6nxz7lrj03mvkkwgvnmp";
   };
   configureFlags = ''
@@ -19,4 +17,10 @@ stdenv.mkDerivation rec {
     --with-mpc=${libmpc}
   '';
   makeFlags = ["CC=clang"];
+  meta = with stdenv.lib; {
+    description = "GNU Fortran compiler, part of the GNU Compiler Collection.";
+    homepage    = "https://gcc.gnu.org/fortran/";
+    license     = licenses.gpl3Plus;
+    platforms   = platforms.darwin;
+  };
 }

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -3676,7 +3676,8 @@ let
     isl = isl_0_14;
   }));
 
-  gfortran = gfortran48;
+  gfortran = if !stdenv.isDarwin then gfortran48
+             else callPackage ../development/compilers/gcc/gfortran-darwin.nix {};
 
   gfortran48 = wrapCC (gcc48.cc.override {
     name = "gfortran";

